test/test_SW_Weather.cc

```diff
@@ -0,0 +1,665 @@
+        );
+        EXPECT_DOUBLE_EQ(
+          climateAverages.PPT_cm,
+          climateOutput.PPT_cm[0]
+        );
+        EXPECT_DOUBLE_EQ(
+          climateAverages.meanTemp_C,
+          climateOutput.meanTemp_C[0]
+        );
+
+        // Climate variables used for C4 grass cover
+        EXPECT_DOUBLE_EQ(
+          climateAverages.minTemp7thMon_C,
+          climateOutput.minTemp7thMon_C[0]
+        );
+        EXPECT_DOUBLE_EQ(
+          climateAverages.frostFree_days,
+          climateOutput.frostFree_days[0]
+        );
+        EXPECT_DOUBLE_EQ(
+          climateAverages.ddAbove65F_degday,
+          climateOutput.ddAbove65F_degday[0]
+        );
+
+        // Climate variables used for cheatgrass cover
+        EXPECT_DOUBLE_EQ(
+          climateAverages.PPT7thMon_mm,
+          climateOutput.PPT7thMon_mm[0]
+        );
+        EXPECT_DOUBLE_EQ(
+          climateAverages.meanTempDriestQtr_C,
+          climateOutput.meanTempDriestQtr_C[0]
+        );
+        EXPECT_DOUBLE_EQ(
+          climateAverages.minTemp2ndMon_C,
+          climateOutput.minTemp2ndMon_C[0]
+        );
+
+
+        EXPECT_NEAR(climateAverages.meanTempMon_C[Jan], -8.432581, tol6);
+        EXPECT_NEAR(climateAverages.maxTempMon_C[Jan], -2.562581, tol6);
+        EXPECT_NEAR(climateAverages.minTempMon_C[Jan], -14.302581, tol6);
+        EXPECT_NEAR(climateAverages.PPTMon_cm[Jan], 15.1400001, tol6);
+        EXPECT_NEAR(climateAverages.PPT_cm, 59.27, tol1);
+        EXPECT_NEAR(climateAverages.meanTemp_C, 4.524863, tol1);
+
+        // Climate variables used for C4 grass cover
+        // (stdev of one value is undefined)
+        EXPECT_NEAR(climateAverages.minTemp7thMon_C, 2.809999, tol6);
+        EXPECT_NEAR(climateAverages.frostFree_days, 92, tol6);
+        EXPECT_NEAR(climateAverages.ddAbove65F_degday, 13.546000, tol6);
+        EXPECT_TRUE(isnan(climateAverages.sdC4[0]));
+        EXPECT_TRUE(isnan(climateAverages.sdC4[1]));
+        EXPECT_TRUE(isnan(climateAverages.sdC4[2]));
+
+        // Climate variables used for cheatgrass cover
+        // (stdev of one value is undefined)
+        EXPECT_NEAR(climateAverages.PPT7thMon_mm, 18.299999, tol6);
+        EXPECT_NEAR(climateAverages.meanTempDriestQtr_C, 0.936387, tol6);
+        EXPECT_NEAR(climateAverages.minTemp2ndMon_C, -12.822068, tol6);
+        EXPECT_TRUE(isnan(climateAverages.sdCheatgrass[0]));
+        EXPECT_TRUE(isnan(climateAverages.sdCheatgrass[1]));
+        EXPECT_TRUE(isnan(climateAverages.sdCheatgrass[2]));
+
+
+        // ------ Reset and deallocate
+        allocDeallocClimateStructs(deallocate, 1, &climateOutput, &climateAverages);
+
+    }
+
+    TEST(ClimateVariableTest, ClimateFromDefaultWeatherSouth) {
+
+        // This test relies on allHist from `SW_WEATHER` being already filled
+        SW_CLIMATE_YEARLY climateOutput;
+        SW_CLIMATE_CLIM climateAverages;
+
+        int deallocate = 0;
+        int allocate = 1;
+
+        Bool inNorthHem = swFALSE;
```

Comment:
could we rename that `Bool inSouthHem = swFALSE`? With `inNorthHem` line 409 etc looks as if this is for the northern hemisphere